### PR TITLE
Feature/2884

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/2884

### CODE REVIEW

File: static-assets/components/cstudio-common/common-api.js [hunk 1]
```diff
@@ -9144,329 +9218,336 @@ CStudioAuthoring.Messages = CStudioAuthoring.Messages || {
 }
 
 CStudioAuthoring.InContextEdit = {
-    registerDialog: function(editorId, context) {
+  messagesSubscription: null,
+
+  messageDialogs: (message) => {
+    amplify.publish('PARENT_FORM_MESSAGES',
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Reformat to 1 line. Change message key to `FORM_ENGINE_MESSAGE_POSTED`.

File: static-assets/components/cstudio-forms/data-sources/child-content.js [hunk 2]
```diff
@@ -15,262 +15,258 @@
...
-    updateItem: function(item, control){
-        if(item.key && item.key.match(/\.xml$/)){
-            var getContentItemCb = {
-                success: function(contentTO) {
-                    item.value =  contentTO.item.internalName || item.value;
-                    control._renderItems();
-                },
-                failure: function() {
-                }
+    // editContent
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/child-content.js]: Move this comment to the `editContent` function.

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 3]
```diff
@@ -20,2761 +20,2898 @@ var parentWindowLocation;
...
-            },
+                  }
+                };
 
-            getModelValue: function(id){
-                var value = null;
-                if(id.indexOf("|") != -1) {
-                    var parts = id.split("|");
-                    var repeatGroup = parts[0];
-
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js]: Do child items also need to subscribe for these messages?

File: static-assets/scripts/host.js [hunk 4]
```diff
@@ -73,7 +73,19 @@
   });
 
   communicator.subscribe(Topics.ICE_ZONE_ON, function (message, scope) {
+    var subscribeCallback = function (_message) {
+      switch (_message.type) {
+        case "FORM_RDY": {
```
Comment (rart) [inline static-assets/scripts/host.js]: Spell it fully (FORM_READY). Also, let's be more specific: `FORM_ENGINE_RENDER_COMPLETE`

File: static-assets/scripts/host.js [hunk 5]
```diff
@@ -73,7 +73,19 @@
...
+          amplify.unsubscribe('PARENT_FORM_MESSAGES', subscribeCallback);
```
Comment (rart) [inline static-assets/scripts/host.js]: Find better name for PARENT_FORM_MESSAGES too please

**Review by rart — CHANGES_REQUESTED** [review]
Missing translations for de, kr. Check all supported languages are covered.

**Review by rart — CHANGES_REQUESTED** [review]
@yureii00 please change the name "Item-specific child content" across the board to be called "Embedded child content"

File: static-assets/components/cstudio-forms/forms-engine.js (hunk 3 above)
Comment (aachavarria) [reply static-assets/components/cstudio-forms/forms-engine.js]: is it in the wrong line number?

File: static-assets/components/cstudio-common/resources/es/base.js [hunk 6]
```diff
@@ -687,7 +687,8 @@ CStudioAuthoring.Messages.registerBundle("contentTypes", "es", {
     publishStarted:"Se ha iniciado el proceso de publicación.",
 
     /*Datasources */
-    childContent: "Contenido Hijos",
+    childContent: "Contenido Hijo",
+    embeddedContent: "Embedded Content",
```
Comment (rart) [inline static-assets/components/cstudio-common/resources/es/base.js]: Not translated. Change to "Contenido Empotrado"

Comment (rart) [reply static-assets/components/cstudio-common/resources/es/base.js]: Doesn't seem like contentType was added to this translations file. Also, still missing the other languages.

Comment (aachavarria) [reply static-assets/components/cstudio-common/resources/es/base.js]: 2: It is already there on the translations.

Comment (aachavarria) [reply static-assets/components/cstudio-common/resources/es/base.js]: updated.

Comment (aachavarria) [reply static-assets/components/cstudio-common/resources/es/base.js]: Added Korean and German translations.

**Review by rart — APPROVED** [review]

**Pull request merged**